ios/gamma/split-view/RNSSplitViewHostController.swift

```diff
@@ -248,21 +252,50 @@ extension RNSSplitViewHostController: RNSSplitViewNavigationControllerViewFrameO
 #endif
 
 extension RNSSplitViewHostController: UISplitViewControllerDelegate {
+  public func splitViewControllerDidCollapse(_ svc: UISplitViewController) {
+    reactEventEmitter.emitOnCollapse()
+  }
+
+  public func splitViewControllerDidExpand(_ svc: UISplitViewController) {
+    reactEventEmitter.emitOnExpand()
+  }
+
+  public func splitViewController(
+    _ svc: UISplitViewController, willHide column: UISplitViewController.Column
+  ) {
+    // TODO: missing impl
+  }
+
   #if compiler(>=6.2)
     public func splitViewController(
       _ svc: UISplitViewController, didHide column: UISplitViewController.Column
     ) {
       if #available(iOS 26.0, *) {
+        // TODO: we may consider removing this logic, because it could be handled by onViewDidDisappear on the column level
+        // On the other hand, maybe dedicated event related to the inspector would be a better approach.
+        // For now I am leaving it, but feel free to drop this method if there's any reason that `onDidDisappear` works better.
         if column != .inspector {
           return
         }
 
         if let inspectorViewController = viewController(for: .inspector) {
           if inspectorViewController.view.window == nil {
-            splitViewHostComponentView.notifyInspectorDidHide()
+            reactEventEmitter.emitOnHideInspector()
```

Comment:
It just brought my attention. Why is check for nullish window needed here?

Reply:
`didHide` might be fired when we're keeping a finger on modal and we cancel dismiss by swipe, therefore this check is needed

Reply:
Let's explain it with code comment. This check w/o additional context information does not make much sense & it would be great to have that inline.

Reply:
https://github.com/software-mansion/react-native-screens/pull/3005/commits/5ce7a84fa591fdcce1ea02c032e8167fb41991ac